### ISSUE TITLE
Address already in use (linux and OS X)

### ISSUE THREAD

**Issue by autopulated** [ARMmbed/valinor#16]
```
> Exception in thread semihost-telnet:
Traceback (most recent call last):
  File "/usr/local/Cellar/python/2.7.9/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/local/Cellar/python/2.7.9/Frameworks/Python.framework/Versions/2.7/lib/python2.7/threading.py", line 763, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/usr/local/lib/python2.7/site-packages/pyOCD/target/semihost.py", line 327, in _server
    self.connected = self._abstract_socket.connect()
  File "/usr/local/lib/python2.7/site-packages/pyOCD/gdbserver/gdb_socket.py", line 36, in connect
    self.init()
  File "/usr/local/lib/python2.7/site-packages/pyOCD/gdbserver/gdb_socket.py", line 31, in init
    self.s.bind(('', self.port))
  File "/usr/local/Cellar/python/2.7.9/Frameworks/Python.framework/Versions/2.7/lib/python2.7/socket.py", line 224, in meth
    return getattr(self._sock,name)(*args)
error: [Errno 48] Address already in use
```

Note that this does not prevent the debug session from working correctly.

**Comment by simonqhughes** [comment]
ARM Internal Ref: IOTSFW-1068

**Comment by autopulated** [comment]
Fixed in pyOCD 0.6.1
